<launch>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="odom"/>
  <arg name="odom_topic" default="odom" />

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find marin_controller)/param/holonomic/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find marin_controller)/param/holonomic/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find marin_controller)/param/holonomic/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find marin_controller)/param/holonomic/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find marin_controller)/param/holonomic/base_local_planner_params.yaml" command="load"/>
    <rosparam file="$(find marin_controller)/param/holonomic/dwa_local_planner_params.yaml" command="load"/>
    <rosparam file="$(find marin_controller)/param/holonomic/move_base_params.yaml" command="load"/>
    <rosparam file="$(find marin_controller)/param/holonomic/global_planner_params.yaml" command="load" />
    <rosparam file="$(find marin_controller)/param/holonomic/navfn_global_planner_params.yaml" command="load" />


    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    
  </node>
</launch>
